packages/extension/src/index.ts

```diff
@@ -45,15 +45,19 @@ const plugin: JupyterFrontEndPlugin<void> = {
 
     const api: ICommandBridgeRemote = {
       async execute(command: string, args: ReadonlyPartialJSONObject) {
-        await commands.execute(command, args);
+        return await commands.execute(command, args);
       },
-      listCommands() {
+      async listCommands() {
         return commands.listCommands();
       }
     };
 
     const endpoint = windowEndpoint(self.parent);
     expose(api, endpoint);
+
+    app.started.then(() => {
+      window.parent?.postMessage('extension-loaded', '*');
```

Comment:
Wondering if `createBridge` could expose a new `ready: Promise<void>`, to make waiting for the extension in the IFrame to be ready more convenient?

Reply:
Also there may be some race conditions depending on when the host defines the event listener and when the IFrame posts the message (too early / late).

Reply:
Looks like there is an open issue about that in Comlink: https://github.com/GoogleChromeLabs/comlink/issues/665

So in our case we can likely implement a workaround by defining custom logic for exposing the `ready` state of the IFrame.

Reply:
Opened #47 to explore adding a `ready` promise.